src/hooks/useEthPrice.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+
+// Uses CoinGecko's free api to get eth price.
+// Caches the result in local storage for 5 minutes.
+export function useEthPrice() {
```

Comment:
The `ethPrice` and `eth_price_expiration` are stored in local storage to cache the eth price. This is to prevent a large and unnecessary number of api calls to CoinGecko. This is a free api that doesn't require a key, but we should look into using an api that has higher rate limits.

Reply:
This logic could be easily repurposed to create a generic `useCachedState` hook if we have a need for it.